projects/commudle-admin/src/app/feature-modules/external-feed/components/external-feed/external-feed.component.ts

```diff
@@ -10,28 +10,97 @@ import { IFeedItem } from 'projects/shared-models/feed-item.model';
 })
 export class ExternalFeedComponent implements OnInit {
   externalPosts: IFeedItem[] = [];
+  externalPostsV2 = [];
+  page = 1;
+  count = 10;
   total;
   isLoading = false;
   canLoadMore = true;
+  tags: Array<string> = [];
+  tagsMap:any = {};
+  tagsChecked = [];
 
   constructor(
     private feedItemService: FeedItemService,
     private title: Title,
-    private meta: Meta
+    private meta: Meta, 
   ) {
   }
 
   ngOnInit() {
-    this.getFeedPosts();
     this.setMeta();
+    this.getPopularTags();
+    this.initTagsMap();
+    // this.getFeedPosts();
+    this.getFeedPostsV2();
   }
 
+  initTagsMap() {
+    for (var x = 0; x<this.tags.length; x++) {
+        this.tagsMap[this.tags[x]] = false;
+    }
+  }
+  
+  getPopularTags() {
+    this.feedItemService.pGetPopularTags().subscribe((value :any)=> {
+      let popularTags = value.tags;
+      console.log(popularTags);
+      for (let index in popularTags) {
+        let popularTag = popularTags[index]['tag'];
+        this.tags.push(popularTag);
+      }
+    }); 
+  }
+  updateCheckedTags(tag, event) {
+   this.page = 1
+   this.tagsMap[tag] = event.target.checked;
+   this.tagsChecked = [];
+   for(var x in this.tagsMap) {
+        if(this.tagsMap[x]) {
+            this.tagsChecked.push(x);
+        }
+    }
+    this.externalPostsV2 = [];
+    this.canLoadMore = true;
+    this.isLoading = false;
+    this.getFeedPostsV2();
+  } 
+
   getFeedPosts(): void{
     this.feedItemService.pGetAll().subscribe(value=> {
       this.externalPosts = value.feed_items;
     });
   }
 
+  getFeedPostsV2(){
+    if (!this.isLoading && (!this.total || this.externalPostsV2.length < this.total)) {
+      if (this.tagsChecked.length == 0) {      
+        this.feedItemService.pGetAllv2(this.page).subscribe((value :any ) =>{
```

Comment:
let's please replace value by data everywhere else, we're using it standard

Reply:
If there's any discrepancy in my code, we're in the process of fixing that too.

Reply:
Replaced